typing/typeclass.ml

```diff
@@ -1299,45 +1326,45 @@ and class_expr_aux cl_num val_env met_env scl =
 (* Approximate the type of the constructor to allow recursive use *)
 (* of optional parameters                                         *)
 
-let var_option = Predef.type_option (Btype.newgenvar ())
+let var_option = Predef.type_option (Btype.newgenvar Type_layout.value)
 
 let rec approx_declaration cl =
   match cl.pcl_desc with
     Pcl_fun (l, _, _, cl) ->
       let arg =
         if Btype.is_optional l then Ctype.instance var_option
-        else Ctype.newvar () in
+        else Ctype.newvar Type_layout.value in
```

Comment:
I'm leaving a CR here and in the other places you suggest in this file, but I also think it would be OK if we punt on any support for unboxed types in the class system until like version 74.